federation/pkg/federation-controller/cluster/clustercontroller.go

```diff
@@ -194,3 +205,14 @@ func (cc *ClusterController) UpdateClusterStatus() error {
 	}
 	return nil
 }
+
+func NewClusterClientset(c *federation_v1alpha1.Cluster) (*clientset.Clientset, error) { // TODO: Stolen from federation/pkg/federation-controller/service/cluster_helper.go - factor it out of there.
+	clusterConfig, err := clientcmd.BuildConfigFromFlags(c.Spec.ServerAddressByClientCIDRs[0].ServerAddress, "")
```

Comment:
Now that the secretRef PR is merged, we should be using that here. Pass kubeconfig here instead of the empty second parameter.